erpnext/accounts/doctype/journal_entry/journal_entry.py

```diff
@@ -383,6 +383,13 @@ def make_gl_entries(self, cancel=0, adv_adj=0):
 		gl_map = []
 		for d in self.get("accounts"):
 			if d.debit or d.credit:
+				account_report_type = frappe.db.get_value("Account", d.account, "report_type")
+				if account_report_type == "Profit and Loss":
+					project_name=d.project_name
+                    			support_ticket=d.support_ticket
+                    		else:
+                    			project_name=''
+                    			support_ticket=''
```

Comment:
You should write this block of code in a common place. May be inside `get_gl_dict()` function.